src/AvoInspector.ts

```diff
@@ -162,14 +275,32 @@ export class AvoInspector {
 
   extractSchema(eventProperties: {
     [propName: string]: any;
-  }): Array<{
+  }, shouldLogIfEnabled = true): Array<{
     propertyName: string;
     propertyType: string;
     children?: any;
   }> {
     try {
       this.sessionTracker.startOrProlongSession(Date.now());
-      return new AvoSchemaParser().extractSchema(eventProperties);
+
+      if (this.avoDeduplicator.hasSeenEventParams(eventProperties, true)) {
+        if (shouldLogIfEnabled && AvoInspector.shouldLog) {
+          console.warn(
+            "Avo Inspector: WARNING! You are trying to extract schema shape that was just reported by your Avo functions. " +
+              "This is an indicator of duplicate inspector reporting. " +
+              "Please reach out to support@avo.app for advice if you are not sure how to handle this."
+          );
```

Comment:
Should we warn about the auto-duplication working?

Reply:
what do you mean?

Reply:
I might be misunderstanding what this is, here is how I understand it: If duplicates are spotted and resolved by the dedupe logic we log out this warning message.

If that is the case I'm wondering if we should be warning about it, I think it's just expected behaviour?

Reply:
This warning is when user extracts a schema using standalone `extractSchema` method and we've recently seen exact same schema reported by avo function.

If we do heavy deduplication here we would return empty schema I guess and it might be confusing. Instead we extract schema as usual, but warn that the same schema was just sent from Avo fucntions.